### PR TITLE
Container Manager Wait Until Ready

### PR DESCRIPTION
**Pull request by chester-leung**
The changes force the `DockerContainerManager` and the `KubernetesContainerManager` to sleep until the container/deployment is ready.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,4 +18,6 @@ _build/
 dependency-reduced-pom.xml
 *.iml
 
+k8s_starter.py
```
Comment (dcrankshaw) [inline .gitignore]: Remove this

File: clipper_admin/clipper_admin/docker/docker_container_manager.py [hunk 2]
```diff
@@ -237,8 +241,17 @@ def set_num_replicas(self, name, version, input_type, image, num_replicas):
                     name=name,
                     version=version,
                     missing=(num_missing)))
+
+            model_container_names = []
             for _ in range(num_missing):
-                self._add_replica(name, version, input_type, image)
+                name = self._add_replica(name, version, input_type, image)
+                model_container_names += [name]
```
Comment (dcrankshaw) [inline clipper_admin/clipper_admin/docker/docker_container_manager.py]: Change this to `model_container_names.append(name)`. The way you're doing it right now constructs a new list in each iteration.

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
Not sure why that test is failing, but other than that this looks good. Just some minor comments.

File: .gitignore (hunk 1 above)
Comment (chester-leung) [reply .gitignore]: Done

Comment (dcrankshaw) [reply .gitignore]: It doesn't look like this was removed?

File: .gitignore [hunk 3]
```diff
@@ -18,4 +18,6 @@ _build/
...
+integration-tests/replica_test.py
```
Comment (dcrankshaw) [inline .gitignore]: Remove this.

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
This is good to go once you remove the changes to .gitignore

File: .gitignore (hunk 3 above)
Comment (chester-leung) [reply .gitignore]: Oops. Must have forgotten to save. Sorry about that.

**Review by dcrankshaw — APPROVED** [review]
lgtm

**Pull request merged**